src/api/rpc_api/state.rs

```diff
@@ -263,6 +284,35 @@ where
 		Ok(storage)
 	}
 
+	async fn get_all_storage_keys_paged_up_to_count(
+		&self,
+		storage_key_prefix: Option<StorageKey>,
+		count: u32,
+		start_key: Option<StorageKey>,
+		at_block: Option<Self::Hash>,
+	) -> Result<Vec<StorageKey>> {
+		let mut storage_keys: Vec<StorageKey> = Vec::new();
+		let mut still_todo = count;
+		let mut new_key = start_key;
+
+		while still_todo > 0 {
```

Comment:
Can't we get stuck in this loop if there are fewer events available than `count`?